src/lib/services/client-metrics/metrics-service-v2.ts

```diff
@@ -150,12 +155,53 @@ export default class ClientMetricsServiceV2 {
 
     async getClientMetricsForToggle(
         toggleName: string,
-        hoursBack?: number,
+        hoursBack: number = 24,
     ): Promise<IClientMetricsEnv[]> {
-        return this.clientMetricsStoreV2.getMetricsForFeatureToggle(
-            toggleName,
-            hoursBack,
+        const hours = generateHourBuckets(hoursBack);
+        const metrics =
+            await this.clientMetricsStoreV2.getMetricsForFeatureToggle(
+                toggleName,
+                hoursBack,
+            );
+
+        const environments = metrics.reduce(
+            (unique, item) =>
+                unique.includes(item.environment)
+                    ? unique
+                    : [...unique, item.environment],
+            [],
         );
+
+        const applications = metrics.reduce(
```

Comment:
I think this is totally fine in it's current state but if you want something that avoids the array assignment/lookups in a loop or something more concise, this should work:
```js
const applications = [...new Set(metrics.map(x => x.appName))];
```